manifests/agent.pp

```diff
@@ -340,10 +356,38 @@
   # to network name. If more than 1 interfaces are available, we
   # can find the ipaddress of this specific interface if listenip
   # is set to for example "eth1" or "bond0.73".
-  $listen_ip = $listenip ? {
-    /^(e|lo|bond|lxc|tap|tun|virbr).*/ => fact("networking.interfaces.${listenip}.ip"),
-    '*' => undef,
-    default => $listenip,
+  $listen_array = flatten([$listenip])
+  $listen_ips = $listen_array.map |$listenip| {
+    $listenip ? {
+      /^(e|lo|bond|lxc|tap|tun|virbr).*/ => fact("networking.interfaces.${listenip}.ip"),
+      '*' => undef,
+      default => $listenip,
+    }
+  }
+  $listen_ip = join($listen_ips,',')
+
+
+  # If the user specified a psk but no file we will use the default filename.
+  # If only the file is specified we configure it.
+  # If neither file nor psk is specified nothing is configured.
+  if $tlspsk and ! $tlspskfile {
+    $real_tlspskfile = $zabbix::params::default_pskfile_path
```

Comment:
what's the benefit of having default_pskfile_path and tlspskfile? can't we just set zabbix::agent::tlspskfile to the value you assigned in params.pp to default_pskfile_path?